### ISSUE TITLE
Allow editor options with editFeatures/drawFeatures

### ISSUE THREAD

**Issue by timelyportfolio** [r-spatial/mapedit#92]
@tim-salabim brings up a good point in email and https://github.com/r-spatial/mapedit/issues/90#issuecomment-459247014 that currently there is no way to provide editor options in `editFeatures` or `drawFeatures`.  This likely most affects `editor = "leafpm"` since a user might want to change `snappable`, `snapDistance`, and `allowSelfIntersection`.  I'd like to target this for next release.

**Comment by tim-salabim** [comment]
Next release sounds good. Let's draw the RConsortium project to a close first

**Comment by lbusett** [comment]
I think being able to tweak also the "appearance" of the editFeatures map would be good, if possibile (e.g., stroke width and color, polygon transparency, shape/size of vertexes while editing....).

**Comment by JoshOBrien** [comment]
I just went ahead and modified `editFeatures` to take and process additional arguments `pmToolbarOpts`, `pmDrawOpts`, `pmEditOpts`, and `pmCutOpts`, which it then passes on to the corresponding **leafpm** functions. Would this be welcome as a pull request, and if so, should I submit that and then link to this Issue in the pull request message?

**Comment by tim-salabim** [comment]
Yes, a PR would be great. You can simply put someting like " addresses #92" in your commit message to link the PR to this issue.
